express/utils.py

```diff
@@ -0,0 +1,30 @@
+"""
+Import utils
+"""
+import logging
+import importlib.util
+import importlib.metadata
+from typing import Sequence, Union, List
+
+logger = logging.getLogger(__name__)
+
+
+def is_module_available(module_name: Union[str, Sequence[str]]) -> List[str]:
```

Comment:
Maybe we can have a single utils folder in express that includes:
- kfp_utils.py
- import_utils.py which includes the soft dependency checkers. This can be done similar to how it's done here: https://github.com/huggingface/transformers/blob/main/src/transformers/utils/import_utils.py

Reply:
> Thanks @PhilippeMoussalli
> 
> This should be an internal function though, so it should not be necessary to use it in the examples. The reason you need this, is because you are still raising an `ImportError` at import time, just with a clearer message. We should instead delay this error until the dependency is actually used.
> 
> See the way it is done in [Connexion](https://github.com/spec-first/connexion/blob/8a85a4fe01d6e3adfae3d886d77832be8726b3c5/connexion/__init__.py#L21) or [HuggingFace](https://discuss.huggingface.co/t/adding-a-new-model-to-transformers-with-additional-dependencies/916).

Thanks for reviewing. Indeed it makes more sense to implement it in the express package. Discussed with @NielsRogge  and decided to go for the current approach: 

- Soft dependencies include HF dataset and pandas. If they are not installed, the use will be asked to install them with pip. Reason we didn't include them in `pyproject.toml` is that it's not worth defining an extra for one packages and both pandas and HF dataset should support backwards compatibility. 
- The other KFP dependencies which are just necessary for compiling and uploading pipelines to KFP are now in the  `express/pipelines` repo under a `requirements.txt`.

Also made some small changes to the functions that check for package availability to make them similar to HF